### PR TITLE
[E1000] Minor improvements

### PR DESCRIPTION
**Pull request by disean**
## Purpose

- Add support for 64-bit counters (Intel gigabit NICs)
- Properly handle OIDs
- Check that the packet filter is not 0
- Performance increase

### CODE REVIEW

File: drivers/network/dd/e1000/hardware.c [hunk 1]
```diff
@@ -53,26 +53,25 @@ static USHORT SupportedDevices[] =
     0x10B5,     // Intel 82546GB Quad Copper KSP3
 };
 
-
-static ULONG E1000WriteFlush(IN PE1000_ADAPTER Adapter)
+static
+inline
```
Comment (Extravert-ir) [inline drivers/network/dd/e1000/hardware.c:57]: `static` and `static inline` have no practical difference, so I'd like to leave only `static`

File: drivers/network/dd/e1000/send.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*
+ * PROJECT:     ReactOS Intel PRO/1000 Driver
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Sending packets
+ * COPYRIGHT:   2018 Mark Jansen (mark.jansen@reactos.org)
+ *              2019 Victor Perevertkin (victor.perevertkin@reactos.org)
```
Comment (Extravert-ir) [inline drivers/network/dd/e1000/send.c:6]:
```suggestion
 * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later.html)
 * PURPOSE:     Sending packets
 * COPYRIGHT:   Copyright 2018 Mark Jansen <mark.jansen@reactos.org>
 *              Copyright 2019 Victor Perevertkin <victor.perevertkin@reactos.org>
```

File: drivers/network/dd/e1000/nic.h [hunk 3]
```diff
@@ -26,6 +26,11 @@
 
 typedef struct _E1000_ADAPTER
 {
+    /* NIC Memory */
+    volatile PUCHAR IoBase;
+    NDIS_PHYSICAL_ADDRESS IoAddress;
+    ULONG IoLength;
```
Comment (Extravert-ir) [inline drivers/network/dd/e1000/nic.h:32]: What's the point in moving up?

**Review by Extravert-ir — CHANGES_REQUESTED** [review]

File: drivers/network/dd/e1000/hardware.c (hunk 1 above)
Comment (disean) [reply drivers/network/dd/e1000/hardware.c:57]: Why? Compiler behaves in different ways for `static` and `static inline`

File: drivers/network/dd/e1000/nic.h (hunk 3 above)
Comment (disean) [reply drivers/network/dd/e1000/nic.h:32]: The MMIO pointer is now at offset 0

**Review by Extravert-ir — APPROVED** [review]

**Pull request merged**